src/core/net/srp_server.cpp

```diff
@@ -722,6 +725,11 @@ Error Server::ProcessZoneSection(const Message &aMessage, MessageMetadata &aMeta
     aMetadata.mOffset = offset;
 
 exit:
+    if (error != kErrorNone)
+    {
+        LogCrit("Failed to process DNS Zone section: %s", ErrorToString(error));
```

Comment:
```suggestion
        LogWarn("Failed to process DNS Zone section: %s", ErrorToString(error));
```
Failing to process a DNS message should not be critical for the whole OT stack?